core/src/main/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequest.java

```diff
@@ -77,7 +80,7 @@
 
     private Settings settings = EMPTY_SETTINGS;
 
-    private Map<String, String> mappings = new HashMap<>();
+    private Map<String, Tuple<XContentType, BytesReference>> mappings = new HashMap<>();
```

Comment:
I think for backwards compatibility we should allow for non JSON right now. We should have a separate issue for JSON only